graylog2-server/src/main/java/org/graylog2/configuration/HttpConfiguration.java

```diff
@@ -86,6 +86,10 @@ public class HttpConfiguration {
     @Parameter(value = "http_allow_embedding")
     private boolean httpAllowEmbedding = false;
 
+    @Parameter(value = "http_enable_tcp_keepalive")
+    private boolean httpEnableTcpKeepAlive = false;
```

Comment:
This has some deeper consequences. We'll have to decide for each notification what type of okHttpClient we need. We can either hold both keep-alive and non-keep-alive versions there (but then our threads configuration won't match) or we'll need to create one client per one notification execution, which will kill the keep-alive functionality anyway. 

Maybe we can take it from the other side and allow to close the connection after the notification is submitted. This would fix the current problem as well, without this complicated logic around keep-alive.

Reply:
hmm, bummer. but at least we would have to rename the config to something `http_notifications_tcp_keepalive`

Not a big fan of turning off the connection pool altogether, just because M$ crappy Azure Firewall :/

Reply:
So in my customer's situation, the packet capture did not see any TCP Keep-Alives from Graylog after the initial Alert data packet was sent. This caused Azure to think the connect was dead and terminate it. Then when Graylog wanted to send another alert it failed of course because the session was closed.

So yes- Azure _**should**_ be sending RST packets to properly close the connections after X time of inactivity, but that issue would be avoided if Graylog was actually sending Keep-Alives, which it does not seem to be.

Reply:
@todvora another alternative would be to put it into the system config (see `EventsConfigurationProvider`)

Reply:
Made it configurable via EventsConfiguration and refactored the usage in notifications, so it accepts the config value and sets the flag accordingly.